### ISSUE TITLE
How can I get current filename from rule.test in replace function?

### ISSUE THREAD

**Issue by Zummer** [Va1/string-replace-loader#66]
```
{
   test: /\.js$/,
   loader: 'string-replace-loader',
  options: {
      search: /__\w{6}\b/g,
      replace: function () {
          return `${current filename? or pathname?}`; // this.resourcePath ?
      },
  },
```

**Comment by Zummer** [comment]
where is this code? I did not find it in project
https://github.com/Va1/string-replace-loader/pull/55/commits/69f4c497ea03b9c4a84ff2f18513a6943931434a

**Comment by Va1** [comment]
@smartin85 's PR merged and released as [v3.1.0](https://github.com/Va1/string-replace-loader/releases/tag/v3.1.0). Replace function now has the context of the loader which allows for getting the filename beyond other things.
